### PR TITLE
added definition for __nonzero__ to ProductSet class

### PR DESCRIPTION
**Pull request by frankwang95**
Defined `__nonzero__` for ProductSet class in response to issue #10859.  Function returns true if every argument in a ProductSet class is nonempty.

### CODE REVIEW

**Comment by asmeurer** [comment]
You should define `__bool__` and set `__nonzero__ = __bool__`, so that it works in Python 2 and 3.

**Comment by asmeurer** [comment]
Add a test.

**Comment by frankwang95** [comment]
I've changed the definition for Python 2 and 3 compatibility and added some basic tests to the test file.

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -665,6 +665,10 @@ def _measure(self):
     def __len__(self):
         return Mul(*[len(s) for s in self.args])
 
+    def __bool__(self):
+        return all([bool(s) for s in self.args])
```
Comment (leosartaj) [inline sympy/sets/sets.py]: Add a newline here.

File: sympy/sets/tests/test_sets.py [hunk 2]
```diff
@@ -1021,3 +1022,13 @@ def test_issue_10285():
     ivl = Interval.Lopen(1, oo)
     assert FiniteSet(eq).intersect(ivl) == \
         FiniteSet(s).intersect(Interval.Lopen(2, oo))
+
+
+
```
Comment (leosartaj) [inline sympy/sets/tests/test_sets.py]: Just two lines should be enough.

Comment (frankwang95) [inline sympy/sets/tests/test_sets.py]: Sorry for that, it should be fixed now.

**Comment by oscarbenjamin** [comment]
I don't see why we would want to define `__bool__` for sets. I think that the tests added here would pass anyway if `__bool__` was not defined since the cases where `__bool__` gives False are those where the set has already evaluated to the EmptySet and isn't a ProductSet any more anyway.

The linked issue seems to be fixed already:
```julia
In [5]: Interval(2,3)*Interval(1,2)                                                                                                            
Out[5]: [2, 3] × [1, 2]

In [6]: bool(_)                                                                                                                                
Out[6]: True
```

Thank you for sending this but I'm going to close this PR now. If you want these changes made then please open a new up to date PR.

**Pull request closed without merging**